packages/vscode-extension/src/plugins/network/NetworkDevtoolsWebviewProvider.ts

```diff
@@ -31,35 +37,20 @@ export class NetworkDevtoolsWebviewProvider implements WebviewViewProvider {
       throw new Error("Couldn't retrieve websocket port from network plugin");
     }
 
-    const webviewBase = webview.asWebviewUri(baseUri);
-    const inspectorJsUri = webview.asWebviewUri(
-      Uri.joinPath(baseUri, "entrypoints", "inspector", "inspector.js")
-    );
-
     webviewView.onDidChangeVisibility(() =>
       reportToolVisibilityChanged(NETWORK_PLUGIN_ID, webviewView.visible)
     );
 
-    webview.html = /* html */ `
-    <!DOCTYPE html>
-    <html lang="en">
-    <base href="${webviewBase}">
-    <meta charset="utf-8">
-    <title>DevTools</title>
-    <style>
-      @media (prefers-color-scheme: dark) {
-        body {
-          background-color: rgb(41 42 45);
-        }
-      }
-    </style>
-    <meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src ${webview.cspSource} https:; connect-src ws://localhost:${wsPort} ${webview.cspSource}; script-src 'unsafe-inline' ${webview.cspSource}; style-src 'unsafe-inline' ${webview.cspSource};">
-    <meta name="referrer" content="no-referrer">
-    <script>
-    window.__websocketEndpoint = 'localhost:${wsPort}';
-    </script>
-    <script type="module" src="${inspectorJsUri}"></script>
-    <body class="undocked" id="-blink-dev-tools"/>
-    </html>`;
+    webview.html = generateWebviewContent(
+      this.context,
+      webviewView.webview,
+      this.context.extensionUri,
+      NETWORK_DEV_SERVER_HOST,
+      PREVIEW_NETWORK_NAME,
+      PREVIEW_NETWORK_PATH
+    ).replace(
+      "</body>",
+      `<script>window.__websocketEndpoint = 'localhost:${wsPort}';</script></body>`
```

Comment:
i'd be cleaner to pass this as argument to `generateWebviewContent`. For example add `extraBodyHTML` parameter and embed that into the generated html.